tests/integration/sms/contacts_interaction_test.js

```diff
@@ -0,0 +1,115 @@
+/* global require, marionette, setup, suite, test */
+'use strict';
+
+var assert = require('chai').assert;
+
+var appRoot = require('app-root-path');
+// TODO Change the path once requireFromApp becomes its own module
+var FromApp = require(appRoot + '/shared/test/integration/require_from_app');
+
+var Messages = FromApp.require('sms', 'lib/messages');
+var InboxView = FromApp.require('sms', 'lib/views/inbox/view');
+var ThreadGenerator = FromApp.require('sms', 'generators/thread');
+var Storage = FromApp.require('sms', 'lib/storage.js');
+
+var Contacts = FromApp.require('contacts', 'lib/contacts');
+var ContactsData = FromApp.require('contacts', 'lib/contacts_data');
+
+marionette('Contacts', function() {
+  var messagesApp;
+
+  var client = marionette.client({
+    prefs: {
+      'focusmanager.testmode': true
+    }
+  });
+
+  var thread;
+  var storage;
+  var contactsData;
+  var inbox;
+
+  var testContact = {
+    givenName: ['FirstName'],
+    familyName: ['LastName'],
+    name: ['FirstName LastName'],
+    tel: [{
+      value: '5551234567',
+      type: ['home']
+    }],
+  };
+
+  setup(function() {
+    var contactsApp = new Contacts(client);
+    contactsApp.launch();
+    contactsData = new ContactsData(client);
+    contactsData.createMozContact(testContact);
+    // Closing the application is needed, in order to be sure to switch to the
+    // right frame when an activity is started
+    contactsApp.close();
```

Comment:
You can add reference to https://bugzilla.mozilla.org/show_bug.cgi?id=1140344 that was trying to get rid of this nasty behavior, but was totally ignored by reviewer :)